libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -72,19 +77,30 @@ AP_GPS_SBF::read(void)
     }
 
     if (gps._auto_config != AP_GPS::GPS_AUTO_CONFIG_DISABLE) {
-        if (_init_blob_index < ARRAY_SIZE(_initialisation_blob)) {
+        if (_init_blob_index < (ARRAY_SIZE(_initialisation_blob) + 1)) {
             uint32_t now = AP_HAL::millis();
-            const char *init_str = _initialisation_blob[_init_blob_index];
-
             if (now > _init_blob_time) {
                 if (now > _config_last_ack_time + 2500) {
                     // try to enable input on the GPS port if we have not made progress on configuring it
                     Debug("SBF Sending port enable");
                     port->write((const uint8_t*)_port_enable, strlen(_port_enable));
                     _config_last_ack_time = now;
                 } else {
-                    Debug("SBF sending init string: %s", init_str);
-                    port->write((const uint8_t*)init_str, strlen(init_str));
+                    char *init_str = nullptr;
+                    if (_init_blob_index == 0) {
+                        UNUSED_RESULT(asprintf(&init_str, "sso, Stream%d, COM%d%s",
```

Comment:
This may not go well if someone increases MAX_RECEIVERS without adding an extra parameter for the com port.

Reply:
Why wouldn't it be fine? The parameters are declared in an array of size `GPS_MAX_RECEIVERS` like all our other instance parameters. I agree it's bad in that you can't change the param in that case, but it won't blow up, and frankly it's exactly like all the other GPS parameters at that point.

Reply:
The current code won't make AP blow up.  I wasn't suggesting that :-)

I was just pointing out that it configure the GPS into a non-working state where it was working before - perhaps as persistent state.

Still, it's such a corner case - a recompilation already being neessary to trigger the problem - that I don't think we ought to waste more time on it.